### PR TITLE
Fix Issue #2823 - Ugly scrollbars in oembed iframes and broken resizing

### LINKED ISSUE #2823: Ugly scrollbars in oembed iframes and broken resizing
I have a contact who's sharing many oembed posts, and I noticed that the height of the `<iframe>` containing it wasn't set very well. Either it was too big and the post would include a large blank space, it it was too small and scrollbars would start to appear.

I noticed there was an oembed iframe resizing function, but so far it was broken and only used when the oembed didn't have a specified height or if the specified height was expressed in percentage. Even after fixing the resizing function, I noticed that it was quite sudden, and that it wasn't removing the iframe scrollbars.

I came up with a few change to make the oembed easier on the eye, and instead of trusting the remote oembed dimensions that rearely are accurate in our usage (or adding 80 arbitrary pixels), the resizer is running for all the oembeds, with a smooth CSS transition to claim the missing space.

### PR DESCRIPTION
**Pull request by Hypolite**
- Fix _resizeIframe function, shorten the timeout between calls
- Simplify the oembed iframe generation code and reduce the minimum
iframe height to 200px
- Add smooth CSS transition for iframe resizing

Fixes #2823.

### CODE REVIEW

File: include/oembed.php [hunk 1]
```diff
@@ -209,25 +209,32 @@ function oembed_format_object($j){
...
-		$height = '300';
-		$resize = 'onload="resizeIframe(this);"';
-	} else
-		$resize = '';
-
-	// try and leave some room for the description line.
-	$height = intval($height) + 80;
-	$width  = intval($width) + 40;
+/**
+ * Generates the iframe HTML for an oembed attachment. Width and height are given
```
Comment (rabuzarus) [inline include/oembed.php]: the doxygen header should have a `@brief` short description

Comment (annando) [inline include/oembed.php]: I shouldn't review pull requests too early in the morning ;-)

Comment (rabuzarus) [inline include/oembed.php]: I also know this problem :-D At least not before drinking two cups of coffee ;-) Oh today I had only one.

**Comment by rabuzarus** [comment]
While this PR is working for normal embedded content resizing is now broken for embedded video links. E.g. `[youtube]YOUTUBE_URL[/youtube]`

**Pull request merged**